openapi3/schema.go

```diff
@@ -1039,6 +1051,32 @@ func (schema *Schema) visitJSONNumber(settings *schemaValidationSettings, value
 		return schema.expectedType(settings, "number, integer")
 	}
 
+	// formats
+	if schema.Format != "" {
+		formatMin := formatMinInt64
+		formatMax := formatMaxInt64
+		switch schema.Format {
+		case "int32":
+			formatMin = formatMinInt32
+			formatMax = formatMaxInt32
+		}
```

Comment:
This wrongly assumes that only 2 formats are possible. Please default `formatMin,formatMat := 0, 0` (for instance) and from there set their values then check against default.

Reply:
I wrongly assumed the default format was int64. Will fix

Reply:
Also will report an error if another format is provided